### PR TITLE
Feature/user recast

### PR DESCRIPTION
**Pull request by zong09**
[POST] users/:id/recast 
[GET] contents/:id/recasted
[POST] /users/:id/quotecast

### CODE REVIEW

**Comment by kittipong-castcle** [comment]
Please use rebasing instead of merging.

**Review by julapong-castcle — APPROVED** [review]

File: apps/users/src/app/app.controller.ts [hunk 1]
```diff
@@ -999,4 +1030,54 @@ export class UserController {
...
+  @Post(':id/quotecast')
+  async quoteContent(
+    @Param('id') id: string,
+    @Body('castcleId') contentId: string,
+    @Body('message') message: string,
+    @Req() req: CredentialRequest
+  ) {
+    const content = await this._getContentIfExist(contentId);
+
+    const user = await this._getUserFromBody(req, id);
```
Comment (kittipong-castcle) [inline apps/users/src/app/app.controller.ts:1072]:
```suggestion
    @Auth() authorizer: Authorizer
  ) {
    authorizer.requestAccessForUser(id);
```

use this to check authority

File: apps/users/src/app/app.controller.ts [hunk 2]
```diff
@@ -999,4 +1030,54 @@ export class UserController {
...
+    const result = await this.contentService.quoteContentFromUser(
+      content,
+      user,
```
Comment (kittipong-castcle) [inline apps/users/src/app/app.controller.ts:1075]:
```suggestion
      authorizer.user,
```

File: libs/database/src/lib/services/content.service.ts [hunk 3]
```diff
@@ -247,6 +247,39 @@ export class ContentService {
...
+  ) => {
+    this.logger.log('get content for delete.');
+    const content = await this._contentModel
+      .findOne({
+        'author.id': mongoose.Types.ObjectId(authorId),
+        'originalPost._id': mongoose.Types.ObjectId(originalPostId)
+      })
+      .exec();
+
+    if (content) {
```
Comment (kittipong-castcle) [inline libs/database/src/lib/services/content.service.ts:268]:
```suggestion
    if (content) return;
```

please use guard return to reduce block scope

File: libs/database/src/lib/services/content.service.ts [hunk 4]
```diff
@@ -247,6 +247,39 @@ export class ContentService {
...
+      if (content.isRecast || content.isQuote) {
+        this.logger.log('delete engagement.');
+        const engagement = await this._engagementModel
+          .findOne({ itemId: content._id })
+          .exec();
+        await engagement.remove();
```
Comment (kittipong-castcle) [inline libs/database/src/lib/services/content.service.ts:274]: Can we use `findOneAndRemove` instead?

**Review by kittipong-castcle — CHANGES_REQUESTED** [review]

**Review by kittipong-castcle — APPROVED** [review]

**Pull request merged**